File: 1626-best-team-with-no-conflicts/1626-best-team-with-no-conflicts.py
class Solution:
    def bestTeamScore(self, scores: List[int], ages: List[int]) -> int:
        comb = sorted(zip(ages, scores))
        dp = [p[1] for p in comb]
        for i, (_, score) in enumerate(comb):
            for j in [j for j in range(i) if comb[j][1] <= score]:
                dp[i] = max(dp[i], dp[j] + score)
        return max(dp)

File: 1626-best-team-with-no-conflicts/1626-best-team-with-no-conflicts.cpp
class Solution {
  public:
      int bestTeamScore(vector<int>& scores, vector<int>& ages) {
          vector<pair<int, int>> comb;
          for (int i = 0; i < scores.size(); i++) {
              comb.push_back({ages[i], scores[i]});
          }
          sort(comb.begin(), comb.end());
          int n = comb.size();
          int maxi = 0;
          vector<int> dp(n);
          for (int i = 0; i < n; i++) {
              dp[i] = comb[i].second;
              maxi = max(maxi, dp[i]);
          }

          for (int i = 0; i < n; i++) {
              for (int j = i - 1; j >= 0; j--) {
                  if (comb[i].second >= comb[j].second) {
                      dp[i] = max(dp[i], comb[i].second + dp[j]);
                  }
              }
              maxi = max(maxi, dp[i]);
          }

          return maxi;
      }
  };